src/zarr/core/metadata/converter/converter_v2_v3.py

```diff
@@ -0,0 +1,256 @@
+    storage_options: dict[str, Any] | None = None,
+    dry_run: bool = False,
+) -> None:
+    """Remove all v2 (.zarray, .zattrs, .zgroup, .zmetadata) or v3 (zarr.json) metadata files from the given Zarr.
+    Note - this will remove metadata files at all levels of the hierarchy (every group and array).
+
+    Parameters
+    ----------
+    store : StoreLike
+        Store or path to directory in file system or name of zip file.
+    zarr_format : ZarrFormat
+        Which format's metadata to remove - 2 or 3.
+    path : str | None, optional
+        The path within the store to open, by default None
+    storage_options : dict | None, optional
+        If the store is backed by an fsspec-based implementation, then this dict will be passed to
+        the Store constructor for that implementation. Ignored otherwise.
+    dry_run : bool, optional
+        Enable a 'dry run' - files that would be deleted are logged, but no files are actually removed.
+    """
+    store_path = await make_store_path(store, mode="r+", storage_options=storage_options)
+    if not store_path.store.supports_deletes:
+        raise ValueError("Store must support deletes to remove metadata")
+
+    if path is None:
+        prefix = ""
+    else:
+        prefix = path
+
+    if zarr_format == 2:
+        metadata_files = [ZARRAY_JSON, ZATTRS_JSON, ZGROUP_JSON, ZMETADATA_V2_JSON]
+    else:
+        metadata_files = [ZARR_JSON]
+
+    awaitables = []
+    async for file_path in store_path.store.list_prefix(prefix):
+        if file_path.split("/")[-1] in metadata_files:
+            logger.info("Deleting metadata at %s", store_path / file_path)
+
+            if not dry_run:
+                awaitables.append((store_path / file_path).delete())
+
+    await asyncio.gather(*awaitables)
+
+
+def _convert_array_metadata(metadata_v2: ArrayV2Metadata) -> ArrayV3Metadata:
+    chunk_key_encoding = V2ChunkKeyEncoding(separator=metadata_v2.dimension_separator)
+
+    codecs: list[Codec] = []
+
+    # array-array codecs
+    if metadata_v2.order == "F":
+        # F is equivalent to order: n-1, ... 1, 0
+        codecs.append(TransposeCodec(order=list(range(len(metadata_v2.shape) - 1, -1, -1))))
+    codecs.extend(_convert_filters(metadata_v2))
+
+    # array-bytes codecs
+    if not isinstance(metadata_v2.dtype, HasEndianness):
+        codecs.append(BytesCodec(endian=None))
+    else:
+        codecs.append(BytesCodec(endian=metadata_v2.dtype.endianness))
+
+    # bytes-bytes codecs
+    bytes_bytes_codec = _convert_compressor(metadata_v2)
+    if bytes_bytes_codec is not None:
+        codecs.append(bytes_bytes_codec)
+
+    return ArrayV3Metadata(
+        shape=metadata_v2.shape,
+        data_type=metadata_v2.dtype,
+        chunk_grid=metadata_v2.chunk_grid,
+        chunk_key_encoding=chunk_key_encoding,
+        fill_value=metadata_v2.fill_value,
+        codecs=codecs,
+        attributes=metadata_v2.attributes,
+        dimension_names=None,
```

Comment:
A thought - it might be nice to give the option of adding dimension names during conversion? That could go in a follow up PR though.